### PR TITLE
merge Geth v1.11.2

### PR DESCRIPTION
**Pull request by sudeepdino008**

### CODE REVIEW

**Comment by rogarcia** [comment]
+1 for develop merge as working stable on node-bsp-1 for the last couple of days

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -3,37 +3,37 @@ name: build-test
...
+      - uses: actions/checkout@v2
 
-    - name: Set up Go
-      uses: actions/setup-go@v2
-      with:
-        go-version: 1.17
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.18
```
Comment (noslav) [inline .github/workflows/build-test.yml:21]: Okay good to know they've updated to 1.18 still behind by one minor version

**Review by noslav — COMMENTED** [review]
Phew!! got through (had to sift) this huge PR!

#### So it seems like these are the major changes to `bsp-geth` (and `geth`) with this upgrade.
* Withdrawals is the major aspect being brought in  
```
type Withdrawal struct {
 	Index     uint64         `json:"index"`          // monotonically increasing identifier issued by consensus layer
 	Validator uint64         `json:"validatorIndex"` // index of validator associated with withdrawal
 	Address   common.Address `json:"address"`        // target address for withdrawn ether
 	Amount    uint64         `json:"amount"`         // value of withdrawal in Gwei
 }
```
* Getting rid of Abi binding support for objc and javac
* Ropsten support has been removed
* Lots of extended support for hardware wallets
* Extra tooling for stateless execution/propagation of tx, blocks, state
 ```
 transition tool    (`t8n`) : a stateless state transition utility
 transaction tool   (`t9n`) : a transaction validation utility
 block builder tool (`b11r`): a block assembler utility aka block runner
 ```
 * Three kinds of `txs` objects transaction types: `LegacyTx`,
 `AccessListTx`, or `DynamicFeeTx`.
 * Support for verkle trees and updated rawdb as a result
 * New LRU cache (with eager, lazy and priority queues) definitions applied to different block caches
 * Lots of code assisting "state pruning"
 * EIPs 1153 (transient storage) and 3860 (limit and metere init code) enabled! 
 * Shanghai upgrade instruction set is included 
 * BLS curves being included in the crypto algos schematic 
 * devp2p extensions for udpv5
 * Refactored runtime evm metrics
 
 #### Also could please outline the steps you took in-order to make this PR as in
 * git checkout local/geth-v1.11.2 (from covalenthq/go-ethereum/develop)
 * git pull origin/go-ethererum tag v1.12.2 onto local branch 
 * git merge/rebase -I  fix conflicts etc.

**Comment by sudeepdino008** [comment]
> Phew!! got through (had to sift) this huge PR!
> 
> #### So it seems like these are the major changes to `bsp-geth` (and `geth`) with this upgrade.
> * Withdrawals is the major aspect being brought in
> 
> ```
> type Withdrawal struct {
>  	Index     uint64         `json:"index"`          // monotonically increasing identifier issued by consensus layer
>  	Validator uint64         `json:"validatorIndex"` // index of validator associated with withdrawal
>  	Address   common.Address `json:"address"`        // target address for withdrawn ether
>  	Amount    uint64         `json:"amount"`         // value of withdrawal in Gwei
>  }
> ```
> 
> * Getting rid of Abi binding support for objc and javac
> * Ropsten support has been removed
> * Lots of extended support for hardware wallets
> * Extra tooling for stateless execution/propagation of tx, blocks, state
> 
> ```
> transition tool    (`t8n`) : a stateless state transition utility
> transaction tool   (`t9n`) : a transaction validation utility
> block builder tool (`b11r`): a block assembler utility aka block runner
> ```
> 
> * Three kinds of `txs` objects transaction types: `LegacyTx`,
>   `AccessListTx`, or `DynamicFeeTx`.
> * Support for verkle trees and updated rawdb as a result
> * New LRU cache (with eager, lazy and priority queues) definitions applied to different block caches
> * Lots of code assisting "state pruning"
> * EIPs 1153 (transient storage) and 3860 (limit and metere init code) enabled!
> * Shanghai upgrade instruction set is included
> * BLS curves being included in the crypto algos schematic
> * devp2p extensions for udpv5
> * Refactored runtime evm metrics
> 
> #### Am I missing anything major from these?
> #### Also could please outline the steps you took in-order to make this PR as in
> * git checkout local/geth-v1.11.2 (from covalenthq/go-ethereum/develop)
> * git pull origin/go-ethererum tag v1.12.2 onto local branch
> * git merge/rebase -I  fix conflicts etc.

That seems to cover it!

I might not have the exact commands...so I'll outline the steps
1. checkout local/geth-1.11.2 from covalenthq/bsp-geth (develop branch)
2. git fetch go-ethereum (go-ethereum remote points to ethereum/go-ethereum)
3. git merge 1.11.2 (this commit - https://github.com/covalenthq/bsp-geth/pull/118/commits/2afd46d0b57d465d983b09c8cd93c668f51574c1) ; and then resolve conflicts 
4. add some more commits resolving linter or build issues.

File: .golangci.yml [hunk 2]
```diff
@@ -12,22 +12,19 @@ run:
 linters:
   disable-all: true
   enable:
-    - deadcode
```
[the remark this answers is not in the document]
Comment (sudeepdino008) [reply .golangci.yml:15]: these were removed upstream. I guess they didn't wanna fix for these linters

**Comment by noslav** [comment]
> > Phew!! got through (had to sift) this huge PR!
> > #### So it seems like these are the major changes to `bsp-geth` (and `geth`) with this upgrade.
> > 
> > * Withdrawals is the major aspect being brought in
> > 
> > ```
> > type Withdrawal struct {
> >  	Index     uint64         `json:"index"`          // monotonically increasing identifier issued by consensus layer
> >  	Validator uint64         `json:"validatorIndex"` // index of validator associated with withdrawal
> >  	Address   common.Address `json:"address"`        // target address for withdrawn ether
> >  	Amount    uint64         `json:"amount"`         // value of withdrawal in Gwei
> >  }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * Getting rid of Abi binding support for objc and javac
> > * Ropsten support has been removed
> > * Lots of extended support for hardware wallets
> > * Extra tooling for stateless execution/propagation of tx, blocks, state
> > 
> > ```
> > transition tool    (`t8n`) : a stateless state transition utility
> > transaction tool   (`t9n`) : a transaction validation utility
> > block builder tool (`b11r`): a block assembler utility aka block runner
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * Three kinds of `txs` objects transaction types: `LegacyTx`,
> >   `AccessListTx`, or `DynamicFeeTx`.
> > * Support for verkle trees and updated rawdb as a result
> > * New LRU cache (with eager, lazy and priority queues) definitions applied to different block caches
> > * Lots of code assisting "state pruning"
> > * EIPs 1153 (transient storage) and 3860 (limit and metere init code) enabled!
> > * Shanghai upgrade instruction set is included
> > * BLS curves being included in the crypto algos schematic
> > * devp2p extensions for udpv5
> > * Refactored runtime evm metrics
> > 
> > #### Am I missing anything major from these?
> > #### Also could please outline the steps you took in-order to make this PR as in
> > 
> > * git checkout local/geth-v1.11.2 (from covalenthq/go-ethereum/develop)
> > * git pull origin/go-ethererum tag v1.12.2 onto local branch
> > * git merge/rebase -I  fix conflicts etc.
> 
> That seems to cover it!
> 
> I might not have the exact commands...so I'll outline the steps
> 
> 1. checkout local/geth-1.11.2 from covalenthq/bsp-geth (develop branch)
> 2. git fetch go-ethereum (go-ethereum remote points to ethereum/go-ethereum)
> 3. git merge 1.11.2 (this commit - [2afd46d](https://github.com/covalenthq/bsp-geth/commit/2afd46d0b57d465d983b09c8cd93c668f51574c1)) ; and then resolve conflicts
> 4. add some more commits resolving linter or build issues.

ah thanks for that; for the commands you used, you can just paste the bash `history` and that should contain everything!

**Pull request merged**